### PR TITLE
A buuunch of stuff

### PR DESCRIPTION
**Pull request by diasdavid**
- [x] start using dist.ipfs.io in favor of go-builder
- [x] zip -> tar.gz
- [x] remove non used features (cli tool), simplifies a lot of things
- [x] update documentation

//cc @jbenet

### CODE REVIEW

**Comment by jbenet** [comment]
(discussed in person.)
todo:
- [ ] create separate module: `go-ipfs-dep` for project dependencies
- [ ] keep `go-ipfs` module for installing the binary
- [ ] still update this module to work with dist.ipfs.io

**Comment by jbenet** [comment]
@diasdavid what needs to happen here? what's stalled?

**Comment by jbenet** [comment]
and every version should be released, too. all of these should work:

```
npm install -g go-ipfs@0.4.0
npm install -g go-ipfs@0.4.1
npm install -g go-ipfs@0.4.2
npm install -g go-ipfs@0.4.3
npm install -g go-ipfs@0.4.4
```

**Comment by jbenet** [comment]
> remove non used features (cli tool), simplifies a lot of things

why? it _is_ used.

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+var gunzip = require('gunzip-maybe')
+var tarFS = require('tar-fs')
+
+module.exports = function (callback) {
+  checkPlatform(goenv) // make sure we can do this.
+
+  // hacky hack hack to work around unpublishability
+  version = version.replace(/-[0-9]+/, '')
+
+  var filename = 'ipfs_v' + version + '_' + goenv.GOOS + '-' + goenv.GOARCH + '.tar.gz'
```
Comment (jbenet) [inline src/index.js]: this doesn't work for windows. go-ipfs uses `.zip` for windows.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        tarFS
+          .extract(installPath)
+          .on('finish', callback)
+        )
+
+  function checkPlatform (goenv) {
+    switch (goenv.GOOS) {
+      case 'darwin':
+      case 'linux':
+      case 'freebsd':
```
Comment (jbenet) [inline src/index.js]: windows builds exist now.

**Comment by diasdavid** [comment]
> remove non used features (cli tool), simplifies a lot of things
jbennet: why? it is used.

This PR was made before `go-ipfs-dep`, it was making npm-go-ipfs the go-ipfs-dep, but then we made it, so not everything makes sense. (date of PR: April 8)

**Comment by jbenet** [comment]
Ok. i'm fixing this. this is breaking people's stuff.

**Comment by jbenet** [comment]
@diasdavid no, this has nothing to do with all that madness. this just has to do with installing the binary, which was broken and forgotten for no good reason.

**Pull request closed without merging**